### PR TITLE
OpenStack Volume and Keypair content

### PR DESCRIPTION
**Pull request by umavanth**
Signed-off-by: Umamaheswari S uma-maheswaris@hp.com

### CODE REVIEW

**Comment by meirwah** [comment]
@tudorlesan please review

File: content/io/cloudslang/openstack/blockstorage/create_openstack_volume.sl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        default: "'8776'"
+    - token
+    - tenant
+    - volume_name
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/create_openstack_volume.sl]: both proxyHost and proxyPort should have their value as default and be overridable false

File: content/io/cloudslang/openstack/blockstorage/create_openstack_volume_flow.sl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#
+# Inputs:
+#   - host - OpenStack machine host
+#   - identity_port - optional - port used for OpenStack authentication - Default: 5000
+#   - blockstorage_port - optional - port used for OpenStack computations - Default: 8776
+#   - username - OpenStack username
+#   - password - OpenStack password
+#   - tenant_name - name of the project on OpenStack
+#   - size - size of the volume to be created
+#   - tenant_name - name of the project on OpenStack
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/create_openstack_volume_flow.sl]: tenant_name is duplicate in documentation

File: content/io/cloudslang/openstack/blockstorage/delete_openstack_volume.sl [hunk 3]
```diff
@@ -0,0 +1,64 @@
+#   (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+#   All rights reserved. This program and the accompanying materials
+#   are made available under the terms of the Apache License v2.0 which accompany this distribution.
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+# Performs a REST call which deletes an OpenStack volume.
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/delete_openstack_volume.sl]: Description should either be "Deletes OpenStack volume" or the create_openstack_volume description should be "Performs a REST call which creates an OpenStack volume.

File: content/io/cloudslang/openstack/blockstorage/delete_openstack_volume.sl [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        default: "'8776'"
+    - token
+    - tenant
+    - volume_id
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/delete_openstack_volume.sl]: again, proxyHost/proxyPort should have default value and overridable: false

File: content/io/cloudslang/openstack/blockstorage/delete_openstack_volume_flow.sl [hunk 5]
```diff
@@ -0,0 +1,108 @@
+#   (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+#   All rights reserved. This program and the accompanying materials
+#   are made available under the terms of the Apache License v2.0 which accompany this distribution.
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+# Deletes an OpenStack volume.
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/delete_openstack_volume_flow.sl]: Authenticates and deletes OpenStack volume? to match the description for create_openstack_volume_flow.sl

File: content/io/cloudslang/openstack/blockstorage/delete_openstack_volume_flow.sl [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+#
+# Inputs:
+#   - host - OpenStack machine host
+#   - identity_port - optional - port used for OpenStack authentication - Default: 5000
+#   - blockstorage_port - optional - port used for OpenStack computations - Default: 8776
+#   - username - OpenStack username
+#   - password - OpenStack password
+#   - tenant_name - name of the project on OpenStack
+#   - volume_name - name of server to delete
+#   - tenant_name - name of the project on OpenStack
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/delete_openstack_volume_flow.sl]: tenant_name duplicate

File: content/io/cloudslang/openstack/blockstorage/get_openstack_volumes.sl [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    - blockstorage_port:
+        default: "'8776'"
+    - token
+    - tenant
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/blockstorage/get_openstack_volumes.sl]: default value/ overridable: false

File: content/io/cloudslang/openstack/keypair/create_openstack_keypair.sl [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        default: "'8774'"
+    - token
+    - tenant
+    - keypair_name
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/keypair/create_openstack_keypair.sl]: default value/ overridable: false

File: content/io/cloudslang/openstack/keypair/create_openstack_keypair.sl [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    - public_key:
+        required: false
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/keypair/create_openstack_keypair.sl]: shouldn't public_key appear in the body of the POST?

File: content/io/cloudslang/openstack/keypair/delete_openstack_keypair.sl [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        default: "'8774'"
+    - token
+    - tenant
+    - keypair_name
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/keypair/delete_openstack_keypair.sl]: default value/ overridable: false for proxyHost/proxyPort

File: content/io/cloudslang/openstack/keypair/get_openstack_keypairs.sl [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    - compute_port:
+        default: "'8774'"
+    - token
+    - tenant
+    - proxy_host:
+        required: false
+    - proxy_port:
+        required: false
+    - proxyHost: "proxy_host if proxy_host else ''"
+    - proxyPort: "proxy_port if proxy_port else ''"
```
Comment (tudorlesan) [inline content/io/cloudslang/openstack/keypair/get_openstack_keypairs.sl]: default value/ overridable: false for proxyHost/proxyPort

**Pull request merged**